solr/core/src/java/org/apache/solr/handler/admin/CollectionsHandler.java

```diff
@@ -1298,19 +1298,25 @@ public Map<String, Object> execute(
           V2ApiUtils.squashIntoSolrResponseWithoutHeader(rsp, addReplicaPropResponse);
           return null;
         }),
-    // XXX should this command support followAliases?
     DELETEREPLICAPROP_OP(
         DELETEREPLICAPROP,
         (req, rsp, h) -> {
-          Map<String, Object> map =
-              copy(
-                  req.getParams().required(),
-                  null,
-                  COLLECTION_PROP,
-                  PROPERTY_PROP,
-                  SHARD_ID_PROP,
-                  REPLICA_PROP);
-          return copy(req.getParams(), map, PROPERTY_PROP);
+          final RequiredSolrParams requiredParams = req.getParams().required();
+          final String propNameToDelete = requiredParams.get(PROPERTY_PROP);
+          final String trimmedPropNameToDelete =
+              propNameToDelete.startsWith(PROPERTY_PREFIX)
+                  ? propNameToDelete.substring(PROPERTY_PREFIX.length())
+                  : propNameToDelete;
+          final DeleteReplicaPropertyAPI deleteReplicaPropertyAPI =
```

Comment:
I don't think I'm the person to introduce new Java constructs to the Solr code base ;-)  I think if the community is up for it, great...   I might advocate that we do a big sweep through so that all the places `var` makes sense, we put that in, so it's not "a bit one way here, a bit the other way there"  And maybe we even have a way of checking that!

Reply:
I have no issue with `var`.  It's not in my muscle-memory so to speak, but no time like the present to start if there's community agreement.

Do we specify a language-level in our gradle build or anything that'd prevent us from using `var` without build changes?

Reply:
We already have code using "var" albeit not much.  Nobody needed permission :-). It was explicitly forbidden by us by technical controls (validate source patterns gradle) but once 9.0 was released (or soon previously?), this control was intentionally lifted so we can use Java 11 stuff.  For some time, we didn't want to use Java 11 stuff on main (which became 9) to ease back ports to 8x.  If hypothetically we wanted to enable Java 17 on main today, we _might_ similarly want to avoid constructs that would make back-ports harder.

Reply:
Ah, ok.  So the "ban" has already been lifted in a sense.  I'll give it a shot here.